gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageGrpcWriteChannel.java

```diff
@@ -348,13 +480,28 @@ public void onNext(Object response) {
       public void onError(Throwable t) {
         Status s = Status.fromThrowable(t);
         String statusDesc = s == null ? "" : s.getDescription();
-        error =
-            new IOException(
-                String.format(
-                    "Caught exception for '%s', while uploading to uploadId %s at writeOffset %d."
-                        + " Status: %s",
-                    resourceId, uploadId, writeOffset, statusDesc),
-                t);
+
+        if (t.getClass() == StatusException.class || t.getClass() == StatusRuntimeException.class) {
+          Code code =
+              t.getClass() == StatusException.class
+                  ? ((StatusException) t).getStatus().getCode()
+                  : ((StatusRuntimeException) t).getStatus().getCode();
+          if (code == Code.DEADLINE_EXCEEDED
```

Comment:
Create a `HashSet` of transient error codes and check if `code` is in this set instead of using many `||` statements.

Reply:
Created a static Set for these codes for maintainability.